### PR TITLE
changes vmalert query function

### PR DESCRIPTION
**Pull request by f41gh7**
for prometheus rules compability its better to use labels as map
it simplifies template evaluation and allow to ignore can't evaluate field error
because map will return default value.
fixes https://github.com/VictoriaMetrics/operator/issues/243

### CODE REVIEW

**Comment by hagen1778** [comment]
Do you know why tests are failing?

File: app/vmalert/datasource/datasource.go [hunk 1]
```diff
@@ -24,6 +24,30 @@ type Metric struct {
 	Value     float64
 }
 
+// MetricSample is used for templating annotations,
+// Labels as map simplifies templates evaluation.
+type MetricSample struct {
```
Comment (hagen1778) [inline app/vmalert/datasource/datasource.go:29]: Should we change `type Metric` instead?

**Comment by f41gh7** [comment]
>.< linter was broken by lastest kubernetes package changes.

File: app/vmalert/datasource/datasource.go (hunk 1 above)
Comment (f41gh7) [reply app/vmalert/datasource/datasource.go:29]: I'm not happy with public type breaking changes.

 Another option, introduce private type for notifier package and make conversion there?

**Review by hagen1778 — APPROVED** [review]
LGTM!

**Pull request merged**